hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -1227,8 +1227,14 @@ def execute(self, state: OTBNState) -> Optional[Iterator[None]]:
                 # There's a pending EDN request. Stall for a cycle.
                 yield
 
-        # At this point, the WSR is ready. Read it, and update wrd with the
-        # result.
+        # At this point, the WSR is ready. Does it have a valid value? (It
+        # might not if this is a sideload key register and keymgr hasn't
+        # provided us with a value). If not, fail with a KEY_INVALID error.
+        if not state.wsrs.has_value_at_idx(self.wsr):
+            state.stop_at_end_of_cycle(ErrBits.KEY_INVALID)
```

Comment:
Feels a little awkward to give `KEY_INVALID` error based upon the more generic `has_value_at_idx` but probably not worth complicating things.

Reply:
I agree, but thought this would be the simplest thing possible. We'll have to come up with something cleverer if we have any other errors in the future (maybe the WSR could return the error bit or something), but let's go with dumb and simple for now!